repose-aggregator/services/service-client/impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImpl.java

```diff
@@ -71,26 +74,49 @@ public AkkaServiceClientImpl(HttpClientService httpClientService) {
         quickFutureCache = CacheBuilder.newBuilder()
                 .expireAfterWrite(FUTURE_CACHE_TTL, FUTURE_CACHE_UNIT)
                 .build();
+    }
+
+    @PreDestroy
+    public void destroy() {
+        // tie this actor system to our bean lifecycle
+        actorSystem.shutdown();
+    }
 
+    @Override
+    public void setConnectionPoolId(String connectionPoolId) {
+        if (StringUtils.equals(this.connectionPoolId, connectionPoolId)) {
+            return;
+        }
 
-        tokenActorRef = actorSystem.actorOf(new Props(new UntypedActorFactory() {
-            public UntypedActor create() {
+        ActorRef oldTokenActorRef = tokenActorRef;
+        this.connectionPoolId = connectionPoolId;
+        createTokenActor(FORCE_CREATE);
 
-                return new AuthTokenFutureActor(serviceClient);
-            }
-        }).withRouter(new RoundRobinRouter(numberOfActors)), "authRequestRouter");
+        if (oldTokenActorRef != null) {
+            actorSystem.stop(oldTokenActorRef);
+        }
     }
 
+    private synchronized void createTokenActor(ForceCreate forceCreate) {
```

Comment:
Yes, previously it did:
.withRouter(new RoundRobinRouter(numberOfActors)), "authRequestRouter");
... where the number of actors was the connection pool size.

I agree this class could stand to be refactored.  I don't think that should happen in this story, though.  I can update the class to remove the lazy initialization and just start off with a null connection pool id.  That would simplify the multi-threading logic.  Does that sound fine?